src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -247,6 +249,13 @@ private static boolean isSameRequester(ConsentArtefact consentDetail, String req
                                                 .thenReturn(consentApprovalResponse(consents)))));
     }
 
+    private Flux<GrantedConsent> validateDate(List<GrantedConsent> grantedConsents) {
+        return Flux.fromIterable(grantedConsents)
+                .filter(grantedConsent -> grantedConsent.getPermission().getDateRange().getToDate() != null &&
+                        grantedConsent.getPermission().getDateRange().getFromDate() != null)
+                .switchIfEmpty(Flux.error(ClientError.invalidDateRange()));
```

Comment:
can you check from should not be a `future`  date?

Reply:
check with Akshatha what are the validations we should do please.

Reply:
> can you check from should not be a `future`  date?

Sure.